### ISSUE TITLE
DeaDBeeF 0.7.2 version support (port to API 1.9)

### ISSUE THREAD

**Issue by Shellcat-Zero** [kuba160/ddb_gui_prompt#2]
I'm running Ubuntu 16.04 and I have both deadbeef and deadbeef-dev installed from the [PPA](https://launchpad.net/~alex-p/+archive/ubuntu/deadbeef).  Here's the error output:
```
% make all                                                                                                                                                                                                                         
gcc -std=gnu99  -fPIC -I /usr/local/include -O2 -Wall -o properties_gen properties_gen.c properties.c cmd_tools.c
properties_gen.c: In function ‘main’:
properties_gen.c:103:2: warning: ignoring return value of ‘freopen’, declared with attribute warn_unused_result [-Wunused-result]
  freopen("props.c", "w", stdout);
  ^
properties_gen.c:195:2: warning: ignoring return value of ‘freopen’, declared with attribute warn_unused_result [-Wunused-result]
  freopen("props.h", "w", stdout);
  ^
cmd_tools.c: In function ‘cmd_tab_complete_ata’:
cmd_tools.c:193:32: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
                 meta = deadbeef->pl_meta_for_key (item, "artist");
                                ^
cmd_tools.c:202:32: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
                 meta = deadbeef->pl_meta_for_key (item, "title");
                                ^
cmd_tools.c:211:32: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
                 meta = deadbeef->pl_meta_for_key (item, "album");
                                ^
cmd_tools.c:219:28: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
             meta = deadbeef->pl_meta_for_key (item, looking_for);
                            ^
cmd_tools.c: In function ‘cmd_tab_complete_meta’:
cmd_tools.c:279:28: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
             meta = deadbeef->pl_meta_for_key (item, meta_s);
                            ^
cmd_tools.c: In function ‘cmd_get_item’:
cmd_tools.c:382:28: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
             meta = deadbeef->pl_meta_for_key (item, "artist");
                            ^
cmd_tools.c:388:28: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
             meta = deadbeef->pl_meta_for_key (item, "title");
                            ^
cmd_tools.c:394:28: error: ‘DB_functions_t {aka struct <anonymous>}’ has no member named ‘pl_meta_for_key’
             meta = deadbeef->pl_meta_for_key (item, "album");
                            ^
Makefile:29: recipe for target 'properties_gen' failed
make: *** [properties_gen] Error 1
```

**Comment by kuba160** [comment]
This plugin requires DeaDBeeF development version, PPA newest is 0.7.2 which is quite old. I'll write that in my todo, but this make take some time as I'm quite busy now. Fix for this would require finding and substibuting `pl_meta_for_key`  function which came in API 1.10.

Edit: inspect `plt_find_meta`, `pl_get_meta` and `pl_get_meta_raw` which could replace this function

**Comment by Shellcat-Zero** [comment]
Thanks, I just downloaded `deadbeef.h` from the repo and dumped it into `/usr/local/include/deadbeef` and now it appears to have compiled successfully:
```
% make all                                                                                                                                                                                      
gcc -std=gnu99  -fPIC -I /usr/local/include -O2 -Wall -o properties_gen properties_gen.c properties.c cmd_tools.c                                                                                      
properties_gen.c: In function ‘main’:                                                                                                                                                                  
properties_gen.c:103:2: warning: ignoring return value of ‘freopen’, declared with attribute warn_unused_result [-Wunused-result]                                                                      
  freopen("props.c", "w", stdout);                                                                                                                                                                     
  ^                                                                                                                                                                                                    
properties_gen.c:195:2: warning: ignoring return value of ‘freopen’, declared with attribute warn_unused_result [-Wunused-result]                                                                      
  freopen("props.h", "w", stdout);                                                                                                                                                                     
  ^                                                                                                                                                                                                    
./properties_gen                                                                                                                                                                                       
properties_gen: got 4 tabs                                                                                                                                                                             
gcc -std=gnu99 -c -fPIC -I /usr/local/include -O2 -Wall prompt.c assets.c cmd.c cmd_tools.c props.c settings.c                                                                                         
gcc -std=gnu99 -shared -fPIC -I /usr/local/include -O2 -Wall -o ddb_gui_prompt.so prompt.o assets.o cmd.o cmd_tools.o props.o settings.o -lreadline
```

**Comment by Shellcat-Zero** [comment]
Ah, so does this actually require the current version of Deadbeef to be compiled from source and installed for it to run?

**Comment by Shellcat-Zero** [comment]
Well, I compiled and installed Deadbeef from source, then re-compiled ddb_gui_prompt (this time no warning messages in the output).  How is ddb_gui_prompt supposed to be invoked/executed once it is compiled and installed?  I have no idea how to run it after executing `sudo make install`.

**Comment by kuba160** [comment]
> Ah, so does this actually require the current version of Deadbeef to be compiled from source and installed for it to run?

The plugin was written for newer Deadbeef version, but it's possible to backport it, which I may do later. There will be a new deadbeef release soon, so I'm not sure if this would be necessary.

> How is ddb_gui_prompt supposed to be invoked/executed once it is compiled and installed?

You can change GUI in 2 ways:
* You can run deadbeef with GTK2/3 UI and change GUI plugin in `Preferences/"Gui / Misc"/"GUI Plugin"`
* Start deadbeef in console using `deadbeef --gui prompt`

To start deadbeef again in GUI mode, run `deadbeef --gui GTK2` (or GTK3). You can theoretically change gui plugin from the prompt with `> settings config gui_plugin string GTK3` and then restarting deadbeef.

**Comment by Shellcat-Zero** [comment]
Thanks, I've got it working fine with `deadbeef --gui prompt`.  I cannot figure out how to add music from the prompt though.

I really appreciate your work on this project, I'm evaluating the feasibility of [this](https://github.com/cmus/cmus/issues/822) feature upgrade for [CMUS](https://cmus.github.io/).  I primarily work/play music where there is no GUI available, so CMUS (and ncurses interfaces) have become my interface of choice.

**Comment by kuba160** [comment]
Adding music is not implemented, however you can use deadbeef internal standard for adding files by running `deadbeef file.mp3` in separate terminal. This will add song to a playlist and maybe clean that playlist (it depends on your settings)

**Comment by kuba160** [comment]
Closing issue since DeaDBeeF 1.8.0 was released which made 0.7.2 (API 1.9) obsolete.
